src/Nethermind/Nethermind.Baseline/Bytes32.cs

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Runtime.InteropServices;
+using Nethermind.Core.Extensions;
+
+namespace Nethermind.Baseline
+{
+    public class Bytes32 : IEquatable<Bytes32>
+    {
+        private const int Length = 32;
+
+        private readonly byte[] _bytes;
+
+        public static Bytes32 Wrap(byte[] bytes)
```

Comment:
What is the point of this method?

Reply:
clearly indicating that the changes to original byte[] will mutate Bytes32 as simply passing t in the constructor could suggest cloning the value to the internal storage